### DISCUSSION TITLE
Navigation Events in ViewModels

### DISCUSSION THREAD

**Discussion by tibitoth** [unoplatform/uno.extensions#2138] [Ideas]
## What would you like to be added:

I don't find any documentation or sample how to handle navigation events (e.g..: `OnNavigatedTo`) in VMs. Other frameworks have `INavigationAware` interface to delegate those events into VMs.

## Why is this needed:
I would like to handle async VM initialization logic which are not ideal in constructors.

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [X] Windows
- [ ] Build tasks

**Comment by nickrandolph** [comment]
Moved this to a discussion so that we can work out what the API would look like
cc @francoistanguay

**Reply by nickrandolph** [reply]
@tibitoth can you give examples of the type of things you want to be able to do in the initialization method. In the past we've discussed this but in most scenarios have found alternative ways to address

**Reply by tibitoth** [reply]
What is the canonical way to fetch a REST API after navigation? I did not find any examples in docs.
